drivers/usb/device/Kconfig

```diff
@@ -26,6 +31,7 @@ config USB_DC_STM32
 	prompt "USB device controller driver for STM32 devices"
 	depends on HAS_STM32CUBE
 	depends on SOC_SERIES_STM32F4X
+	select USB_DEVICE_DRIVER
```

Comment:
Is using of `select` together with `prompt` and `depends` acceptable?

Reply:
From https://www.kernel.org/doc/Documentation/kbuild/kconfig-language.txt:
>	Select should be used with care. Select will force a symbol to a value without visiting the dependencies.
	By abusing select you are able to select a symbol FOO even if FOO depends on BAR that is not set.
	In general use select only for non-visible symbols (no prompts anywhere) and for symbols with no dependencies. That will limit the usefulness but on the other hand avoid the illegal configurations all over.

I think it is safe here. The USB_DEVICE_DRIVER is non-visible symbol, with no dependencies.